src/components/PlayerSubmissionForm.js

```diff
@@ -3,30 +3,68 @@ import PropTypes from 'prop-types';
 
 import './PlayerSubmissionForm.css';
 
-const PlayerSubmissionForm = () => {
-  return (
-    <div className="PlayerSubmissionForm">
-      <h3>Player Submission Form for Player #{  }</h3>
-
-      <form className="PlayerSubmissionForm__form" >
-
-        <div className="PlayerSubmissionForm__poem-inputs">
-
-          {
-            // Put your form inputs here... We've put in one below as an example
-          }
-          <input
-            placeholder="hm..."
-            type="text" />
-
-        </div>
-
-        <div className="PlayerSubmissionForm__submit">
-          <input type="submit" value="Submit Line" className="PlayerSubmissionForm__submit-btn" />
-        </div>
-      </form>
-    </div>
-  );
+const PlayerSubmissionForm = (props) => {
+
+  const [formFields, setFormFields] = useState(props.fields)
+
+  const onFormFieldChange = (i, event) => {
+    const newFormFields = [
+      ...formFields,
+    ]
+
+    newFormFields[i] = {
+      ...newFormFields[i],
+      userInput: event.target.value
+    };
+
+    setFormFields(newFormFields);
+  }
+
+  const onFormSubmit = (event) => {
+    event.preventDefault();
+
+    props.sendSubmission(formFields);
+    setFormFields(props.fields);
+  };
+
+  const playerComponents = formFields.map((field, i) => {
+    if(typeof(field) === 'object') {
+      return (
+        <input
+        key={field.key}
+        placeholder={field.placeholder}
+        value={field.userInput}
+        name={field.key}
```

Comment:
I think the original values you're assigning the input fields are initially undefined here, so you're getting a warning.